From bba51ecf18ec77b8c3c4ed39bfd833184af9b26a Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Wed, 20 Nov 2024 18:01:04 +0100
Subject: [PATCH 741/841] feat(st): add RNG minor version

Some specific configurations (NIST/HTCR) can depend on the RNG IP minor
version used.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 1c5cf499986611fa51b11825e09f0abe03f8896c feat(st): add RNG minor version]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: I3608bd5cad77616bf0c031c66a8312b65d3e68c5
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/419891
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp1/platform.mk | 4 ++++
 plat/st/stm32mp2/platform.mk | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/plat/st/stm32mp1/platform.mk b/plat/st/stm32mp1/platform.mk
index d6cc67274..0e4540b34 100644
--- a/plat/st/stm32mp1/platform.mk
+++ b/plat/st/stm32mp1/platform.mk
@@ -99,10 +99,12 @@ WORKAROUND_CVE_2022_23960:=	0
 ifeq ($(STM32MP13),1)
 STM32_HASH_VER		:=	4
 STM32_RNG_VER		:=	4
+STM32_RNG_VER_MINOR	:=	2
 STM32_SAES_VER		:=	48 # 0x30
 else # Assuming STM32MP15
 STM32_HASH_VER		:=	2
 STM32_RNG_VER		:=	2
+STM32_RNG_VER_MINOR	:=	1
 STM32_SAES_VER		:=	0 # Unavailable in this platform
 endif
 
@@ -186,6 +188,7 @@ $(eval $(call assert_numerics,\
 		STM32_HASH_VER \
 		STM32_HEADER_VERSION_MAJOR \
 		STM32_RNG_VER \
+		STM32_RNG_VER_MINOR \
 		STM32_SAES_VER \
 		STM32_TF_A_COPIES \
 )))
@@ -201,6 +204,7 @@ $(eval $(call add_defines,\
 		STM32_HASH_VER \
 		STM32_HEADER_VERSION_MAJOR \
 		STM32_RNG_VER \
+		STM32_RNG_VER_MINOR \
 		STM32_SAES_VER \
 		STM32_TF_A_COPIES \
 		STM32MP_CRYPTO_ROM_LIB \
diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 3d4253fca..30f89d4f0 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -88,6 +88,12 @@ PKA_USE_BRAINPOOL_P256T1 	?=	0
 
 STM32_HASH_VER			:=	4
 STM32_RNG_VER			:=	4
+ifeq ($(STM32MP21),1)
+STM32_RNG_VER_MINOR		:=	4
+else
+STM32_RNG_VER_MINOR		:=	3
+endif
+
 STM32_SAES_VER			:=	50 # 0x32
 
 ifeq ($(STM32MP21),1)
@@ -205,6 +211,7 @@ $(eval $(call assert_numerics,\
 		PLAT_PARTITION_MAX_ENTRIES \
 		STM32_HASH_VER \
 		STM32_RNG_VER \
+		STM32_RNG_VER_MINOR \
 		STM32_SAES_VER \
 		STM32_TF_A_COPIES \
 )))
@@ -219,6 +226,7 @@ $(eval $(call add_defines,\
 		PLAT_TBBR_IMG_DEF \
 		STM32_HASH_VER \
 		STM32_RNG_VER \
+		STM32_RNG_VER_MINOR \
 		STM32_SAES_VER \
 		STM32_SAES_CRYP2 \
 		STM32_TF_A_COPIES \
-- 
2.47.2

